src/shared/utils.js

```diff
@@ -0,0 +1,163 @@
+                "{arguments}.1"  // eventArguments
+            ]
+        },
+        clear: {
+            funcName: "gpii.app.timer.clear",
+            args: ["{that}"]
+        },
+        isActive: {
+            funcName: "fluid.isValue",
+            args: "{that}.timer"
+        }
+    }
+});
+
+/**
+ * Starts a timer. In `timeoutDuration` milliseconds the `onTimerFinished`
+ * event will be fired. Any previously registered timers will be cleared
+ * upon the invocation of this function.
+ * @param {Component} that -The `gpii.app.timer` instance.
+ * @param {Number} timeoutDuration -The timeout duration in milliseconds.
+ * @param {Any[]} eventArguments - Events to be passed with the fired event.
+ */
+gpii.app.timer.start = function (that, timeoutDuration, eventArguments) {
+    var timeoutArgs = [
+        that.events.onTimerFinished.fire,
+        timeoutDuration
+    ]
+        .concat(eventArguments);
+
+    that.clear();
+    that.timer = setTimeout.apply(null, timeoutArgs);
+};
+
+/**
+ * Clears the timer (if any).
+ * @param {Component} that -The `gpii.app.timer` instance.
+ */
+gpii.app.timer.clear = function (that) {
+    if (that.timer) {
+        clearTimeout(that.timer);
+        that.timer = null;
+    }
+};
+
+fluid.registerNamespace("gpii.app.applier");
+
+/**
+ * Replaces the value at a given path in a component's model in a single
+ * transaction. Useful because the default behavior when invoking an
+ * `applier.change` function is to merge the new object with the value of
+ * the old object at the specified model path which may not always be
+ * appropriate for all situations.
+ * @param {ChangeApplier} applier - The change applier for a model component.
+ * @param {String} path - The path in the component's model.
+ * @param {Any} value - The new value which should be set at the specified
+ * `path`.
+ * @param {String} [source] - The source of the model change.
+ */
+gpii.app.applier.replace = function (applier, path, value, source) {
+    var transaction = applier.initiate();
+
+    // Remove the previous value at the given path
+    transaction.fireChangeRequest({
+        path: path,
+        type: "DELETE",
+        source: source
+    });
+
+    // Add the new value for the corresponding path
+    transaction.fireChangeRequest({
+        path: path,
+        value: value,
+        type: "ADD",
+        source: source
+    });
+
+    transaction.commit();
+};
+
+/**
+ * A namespace for utility functions related to setting groups.
+ */
+fluid.registerNamespace("gpii.app.settingGroups");
+
+/**
+ * Determines if there is at least one setting group in the provided array of
+ * setting groups which contains at least one setting.
+ * @param {module:gpiiConnector.SettingGroup[]} settingGroups - An array of
+ * setting groups.
+ * @return {Boolean} `true` - if there is at least one setting group with at
+ * least one setting in it.
+ */
+gpii.app.settingGroups.hasSettings = function (settingGroups) {
```

Comment:
This function is unused

Reply:
I guess, you mean untested? Because it is actually used here (even though this is yet another untested piece of code): https://github.com/danailbd/gpii-app/blob/feature/QSS1.1/src/main/surveys/surveyTriggerManager.js#L417.

This brings up another thing - in order to test some of the PSP/QSS functionalities, it will be good if we can have a profile without any settings. Is there such a user profile? If not, can one be created or can we mock such if not?

Reply:
I don't believe there is such a profile, please issue a small pull to universal if you would like

Reply:
You just need to do that, but also remember to rerun the dbData build process which is attached to npm install